src/hooks/useRepsSync.tsx

```diff
@@ -0,0 +1,22 @@
+import { useEffect } from 'react';
+
+type UseRepsSyncProps = {
+    isCurrent: boolean;
+    isFinished: boolean;
+    totalSeconds: number;
+    secondsPassed: number;
+    oneRoundSeconds: number;
+    setRepsRemaining: (seconds: number) => void;
+};
+
+export const useRepsSync = ({ isCurrent, isFinished, totalSeconds, secondsPassed, oneRoundSeconds, setRepsRemaining }: UseRepsSyncProps) => {
+    useEffect(() => {
```

Comment:
you are keeping track of some state variables unnecessarily, which complicates your business logic. If you think about it, all you should really track is `isRunning` and `totalSeconds`. All the others can be calculated dynamicall. For example, knowing the `totalSeconds` you should be able to tell what timer is running, what round its currently in and whether its work / rest, etc.